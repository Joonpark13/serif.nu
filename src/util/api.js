/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
