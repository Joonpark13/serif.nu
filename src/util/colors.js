export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
