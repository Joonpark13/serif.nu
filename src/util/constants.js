/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
