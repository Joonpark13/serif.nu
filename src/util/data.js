import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
