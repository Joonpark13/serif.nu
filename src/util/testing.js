/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
