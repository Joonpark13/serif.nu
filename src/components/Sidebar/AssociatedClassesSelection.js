import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';
import { getFormattedClassSchedule } from 'util/time';
import SidebarHeader from './SidebarHeader';

export const styles = {
  sectionsRoot: {
    padding: 16,
  },
  title: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  divider: {
    marginTop: 5,
  },
};

function AssociatedClassesSelection({
  currentCourseName,
  currentSectionNumber,
  associatedClasses,
  back,
  addSectionWithAssociatedClass,
  classes,
}) {
  return (
    <div className={classes.sectionsRoot}>
      <SidebarHeader title={`${currentCourseName}-${currentSectionNumber}`} back={back} />

      <List>
        {associatedClasses.map(associatedClass => (
          <ListItem
            key={JSON.stringify(associatedClass.schedule)}
            button
            onClick={() => addSectionWithAssociatedClass(associatedClass)}
          >
            <ListItemText>
              <Typography variant="h6">
                {associatedClass.type}
              </Typography>

              <Typography>
                {getFormattedClassSchedule(associatedClass.schedule)}
              </Typography>
              <Typography>{associatedClass.schedule.location}</Typography>
            </ListItemText>
          </ListItem>
        ))}
      </List>
    </div>
  );
}

AssociatedClassesSelection.propTypes = {
  currentCourseName: PropTypes.string.isRequired,
  currentSectionNumber: PropTypes.string.isRequired,
  associatedClasses: PropTypes.arrayOf(PropTypes.object).isRequired,
  back: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  addSectionWithAssociatedClass: PropTypes.func.isRequired,
};

export { AssociatedClassesSelection as UnstyledAssociatedClassesSelection };
export default withStyles(styles)(AssociatedClassesSelection);
