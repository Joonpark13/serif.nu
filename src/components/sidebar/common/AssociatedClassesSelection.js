import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';
import { List, ListItem, ListItemText, Typography } from '@material-ui/core';
import { getFormattedClassSchedule, isUnscheduled } from 'util/time';
import SidebarHeader from '../SidebarHeader';

const useStyles = makeStyles({
  sectionsRoot: {
    padding: 16,
  },
  title: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  divider: {
    marginTop: 5,
  },
});

export default function AssociatedClassesSelection({
  currentCourseName,
  currentSectionNumber,
  associatedClasses,
  back,
  addSectionWithAssociatedClass,
  associatedClassHover,
  associatedClassHoverOff,
}) {
  const classes = useStyles();
  return (
    <div className={classes.sectionsRoot}>
      <SidebarHeader title={`${currentCourseName}-${currentSectionNumber}`} back={back} />

      <List>
        {associatedClasses.map((associatedClass) => {
          const associatedClassIsUnscheduled = isUnscheduled(associatedClass.schedule);
          return (
            <ListItem
              key={JSON.stringify(associatedClass.schedule)}
              button
              onClick={() => addSectionWithAssociatedClass(associatedClass)}
              onMouseEnter={() => associatedClassHover(associatedClass)}
              onMouseLeave={associatedClassHoverOff}
              disabled={associatedClassIsUnscheduled}
            >
              <ListItemText>
                <Typography variant="h6">
                  {associatedClass.type}
                </Typography>

                <Typography color={associatedClassIsUnscheduled ? 'error' : undefined}>
                  {getFormattedClassSchedule(associatedClass.schedule)}
                </Typography>
                <Typography>{associatedClass.schedule.location}</Typography>
              </ListItemText>
            </ListItem>
          );
        })}
      </List>
    </div>
  );
}

AssociatedClassesSelection.propTypes = {
  currentCourseName: PropTypes.string.isRequired,
  currentSectionNumber: PropTypes.string.isRequired,
  associatedClasses: PropTypes.arrayOf(PropTypes.object).isRequired,
  back: PropTypes.func.isRequired,
  addSectionWithAssociatedClass: PropTypes.func.isRequired,
  associatedClassHover: PropTypes.func.isRequired,
  associatedClassHoverOff: PropTypes.func.isRequired,
};
