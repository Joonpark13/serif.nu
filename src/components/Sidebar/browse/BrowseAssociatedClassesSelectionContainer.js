import { connect } from 'react-redux';
import toJS from 'util/to-js';
import {
  addSectionWithAssociatedClassFromBrowse,
  changeBrowseLevel,
  associatedClassHover,
  associatedClassHoverOff,
} from 'actions';
import { selectedCourseNameSelector } from 'selectors';
import AssociatedClassesSelection from '../common/AssociatedClassesSelection';

export function selectedSectionNumberSelector(state) {
  return state.getIn(['browse', 'selected', 'section', 'sectionNumber']);
}

export function selectedSectionAssociatedClassesSelector(state) {
  return state.getIn(['browse', 'selected', 'section', 'associatedClasses']);
}

/* istanbul ignore next */
function mapStateToProps(state) {
  return {
    currentCourseName: selectedCourseNameSelector(state),
    currentSectionNumber: selectedSectionNumberSelector(state),
    associatedClasses: selectedSectionAssociatedClassesSelector(state),
  };
}

/* istanbul ignore next */
const mapDispatchToProps = {
  back: () => changeBrowseLevel('section'),
  addSectionWithAssociatedClass: addSectionWithAssociatedClassFromBrowse,
  associatedClassHover,
  associatedClassHoverOff,
};

export default connect(mapStateToProps, mapDispatchToProps)(toJS(AssociatedClassesSelection));
