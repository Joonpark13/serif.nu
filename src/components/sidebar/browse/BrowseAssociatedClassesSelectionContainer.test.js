import { mockUseSelector, wrapperCreator, mockUseDispatch } from 'util/testing';
import {
  changeBrowseLevel,
  addSectionWithAssociatedClassFromBrowse,
  associatedClassHover,
  associatedClassHoverOff,
} from 'actions';
import AssociatedClassesSelection from '../common/AssociatedClassesSelection';
import BrowseAssociatedClassesSelectionContainer from './BrowseAssociatedClassesSelectionContainer';

describe('BrowseAssociatedClassesSelectionContainer', () => {
  const getWrapper = wrapperCreator(BrowseAssociatedClassesSelectionContainer);
  let dispatchMock;

  beforeEach(() => {
    mockUseSelector('101-0', '12345', []);
    dispatchMock = mockUseDispatch();
  });

  it('renders correctly', () => {
    const wrapper = getWrapper();

    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('changes browse level when navigating back', () => {
    const wrapper = getWrapper();
    wrapper.find(AssociatedClassesSelection).props().back();

    expect(dispatchMock).toHaveBeenCalledWith(changeBrowseLevel('section'));
  });

  it('adds section with associated class', () => {
    const wrapper = getWrapper();
    const associatedClass = {};
    wrapper.find(AssociatedClassesSelection).props().addSectionWithAssociatedClass(associatedClass);

    expect(dispatchMock).toHaveBeenCalledWith(
      addSectionWithAssociatedClassFromBrowse(associatedClass),
    );
  });

  it('shows hovered associated class', () => {
    const wrapper = getWrapper();
    const associatedClass = {};
    wrapper.find(AssociatedClassesSelection).props().associatedClassHover(associatedClass);

    expect(dispatchMock).toHaveBeenCalledWith(
      associatedClassHover(associatedClass),
    );
  });

  it('hides unhovered associated class', () => {
    const wrapper = getWrapper();
    wrapper.find(AssociatedClassesSelection).props().associatedClassHoverOff();

    expect(dispatchMock).toHaveBeenCalledWith(associatedClassHoverOff());
  });
});
