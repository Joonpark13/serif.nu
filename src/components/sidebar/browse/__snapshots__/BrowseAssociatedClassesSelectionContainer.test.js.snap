// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseAssociatedClassesSelectionContainer renders correctly 1`] = `
<AssociatedClassesSelection
  addSectionWithAssociatedClass={[Function]}
  associatedClassHover={[Function]}
  associatedClassHoverOff={[Function]}
  associatedClasses={Array []}
  back={[Function]}
  currentCourseName="101-0"
  currentSectionNumber="12345"
/>
`;
